recipe/build_common.sh

```diff
@@ -0,0 +1,268 @@
+    # because it requires an non-bundled compile-time dependency (libuv
+    # through gloo). This dependency is made available through meta.yaml, so
+    # we can override the default and set USE_DISTRIBUTED=1.
+    export USE_DISTRIBUTED=1
+
+    if [[ "$target_platform" == "osx-arm64" ]]; then
+        # MKLDNN did not support on Apple M1 at the time support Apple M1
+        # was added. Revisit later
+        export USE_MKLDNN=0
+    fi
+elif [[ ${cuda_compiler_version} != "None" ]]; then
+    if [[ "$target_platform" == "linux-aarch64" ]]; then
+        # https://github.com/pytorch/pytorch/pull/121975
+        # https://github.com/conda-forge/pytorch-cpu-feedstock/issues/264
+        export USE_PRIORITIZED_TEXT_FOR_LD=1
+    fi
+    # Even though cudnn is used for CUDA builds, it's good to enable
+    # for MKLDNN for CUDA builds when CUDA builds are used on a machine
+    # with no NVIDIA GPUs.
+    export USE_MKLDNN=1
+    export USE_CUDA=1
+    export USE_CUFILE=1
+    # PyTorch has multiple different bits of logic finding CUDA, override
+    # all of them.
+    export CUDAToolkit_BIN_DIR=${BUILD_PREFIX}/bin
+    export CUDAToolkit_ROOT_DIR=${PREFIX}
+    if [[ "${target_platform}" != "${build_platform}" ]]; then
+        export CUDA_TOOLKIT_ROOT=${PREFIX}
+    fi
+    case ${target_platform} in
+        linux-64)
+            export CUDAToolkit_TARGET_DIR=${PREFIX}/targets/x86_64-linux
+            ;;
+        linux-aarch64)
+            export CUDAToolkit_TARGET_DIR=${PREFIX}/targets/sbsa-linux
+            ;;
+        *)
+            echo "unknown CUDA arch, edit build.sh"
+            exit 1
+    esac
+    case ${cuda_compiler_version} in
+        12.6)
+            export TORCH_CUDA_ARCH_LIST="5.0;6.0;6.1;7.0;7.5;8.0;8.6;8.9;9.0+PTX"
+            ;;
+        *)
+            echo "unsupported cuda version. edit build.sh"
+            exit 1
+    esac
+    export TORCH_NVCC_FLAGS="-Xfatbin -compress-all"
+    export NCCL_ROOT_DIR=$PREFIX
+    export NCCL_INCLUDE_DIR=$PREFIX/include
+    export USE_SYSTEM_NCCL=1
+    export USE_STATIC_NCCL=0
+    export USE_STATIC_CUDNN=0
+    export MAGMA_HOME="${PREFIX}"
+    # Perform the initial build without magma enabled, we'll enable
+    # it for the remaining builds (particularly, to have it enabled
+    # for pytorch).
+    export USE_MAGMA=0
+else
+    if [[ "$target_platform" != *-64 ]]; then
+      # Breakpad seems to not work on aarch64 or ppc64le
+      # https://github.com/pytorch/pytorch/issues/67083
+      export USE_BREAKPAD=0
+    fi
+    # MKLDNN is an Apache-2.0 licensed library for DNNs and is used
+    # for CPU builds. Not to be confused with MKL.
+    export USE_MKLDNN=1
+    export USE_CUDA=0
+fi
+
+echo '${CXX}'=${CXX}
+echo '${PREFIX}'=${PREFIX}
+
+case ${PKG_NAME} in
+  libtorch-split)
+    # Call setup.py directly to avoid spending time on unnecessarily
+    # packing and unpacking the wheel.
+    $PREFIX/bin/python setup.py build
```

Comment:
Calling `setup.py` directly is deprecated right?

Reply:
Depends on who you ask. setuptools itself only deprecated `setup.py install` so far. Then, the hacks PyTorch do on top of setuptools are all pretty much incompatible with PEP517 (especially `sys.argv` manipulations).